marsha/core/models.py

```diff
@@ -0,0 +1,392 @@
+        to=Organization,
+        related_name="playlists",
+        # don't delete a playlist if its organization is deleted
+        on_delete=models.SET_NULL,
+        null=True,
+    )
+    author: User = models.ForeignKey(
+        to=User,
+        related_name="created_playlists",
+        # don't delete a playlist if its author is deleted
+        on_delete=models.SET_NULL,
+        null=True,
+    )
+    editors: M2MType[User] = models.ManyToManyField(
+        to=User,
+        through="PlaylistRight",
+        related_name="managed_playlists",
+        verbose_name=_("editors"),
+        help_text=_("users allowed to manage this playlist"),
+    )
+    is_public: bool = models.BooleanField(
+        default=False,
+        verbose_name=_("is public"),
+        help_text=_("if this playlist can be viewed without any access control"),
+    )
+    duplicated_from: "Playlist" = models.ForeignKey(
+        to="self",
+        related_name="duplicates",
+        verbose_name=_("duplicate from"),
+        help_text=_("original playlist this one was duplicated from"),
+        # don't delete a playlist if the one it was duplicated from is deleted
+        on_delete=models.SET_NULL,
+        null=True,
+    )
+    videos: M2MType[Video] = models.ManyToManyField(
+        to=Video,
+        through="PlaylistVideo",
+        related_name="playlists",
+        verbose_name=_("videos"),
+        help_text=_("videos in this playlist"),
+    )
+
+    # `related_name` of fields pointing to this model, for typing
+    duplicates: ReverseFKType["Playlist"]
+    videos_links: ReverseFKType["PlaylistVideo"]
+    users_rights: ReverseFKType["PlaylistRight"]
+
+    class Meta:
+        """Options for the ``Playlist`` model."""
+
+        verbose_name: str = _("playlist")
+        verbose_name_plural: str = _("playlists")
+
+
+class PlaylistVideo(BaseModel):
+    """Model representing a video in a playlist."""
+
+    video: Video = models.ForeignKey(
+        to=Video,
+        related_name="playlists_links",
+        verbose_name=_("video"),
+        help_text=_("video contained in this playlist"),
+        # delete the video-playlist link if the video is deleted
+        on_delete=models.CASCADE,
+    )
+    playlist: Playlist = models.ForeignKey(
+        to=Playlist,
+        related_name="videos_links",
+        verbose_name=_("playlist"),
+        help_text=_("playlist containing this video"),
+        # delete the video-playlist link if the playlist is deleted
+        on_delete=models.CASCADE,
+    )
+    order: int = models.PositiveIntegerField(
+        verbose_name=_("order"), help_text=_("video order in the playlist")
+    )
+
+    class Meta:
+        """Options for the ``PlaylistVideo`` model."""
+
+        verbose_name: str = _("playlist video link")
+        verbose_name_plural: str = _("playlist video links")
+        unique_together: UniqueTogether = [("video", "playlist")]
+
+
+class PlaylistRight(BaseModel):
```

Comment:
Why not naming it `UserPlaylist` or something similar? Not sure `FooRight` is the way to go.

Reply:
As proposed by sam, `PlaylistAccess` seems good to me